controllers/imagerepository_controller.go

```diff
@@ -518,14 +512,22 @@ func (r *ImageRepositoryReconciler) EnsureRemotePullSecret(ctx context.Context,
 						{
 							ServiceAccount: remotesecretv1beta1.ServiceAccountLink{
 								Reference: corev1.LocalObjectReference{
-									Name: defaultServiceAccountName,
+									Name: serviceAccountName,
 								},
 							},
 						},
 					},
 				},
 			},
 		}
+
+		if isPull {
```

Comment:
also don't understand why this differs for pull & push secret

Reply:
Because push tokne must be in current (build) namespace only (at least for current requirements), but pull token in all environments of the application.